ci-operator/step-registry/baremetalds/metallb-e2e/baremetalds-metallb-e2e-workflow.yaml

```diff
@@ -0,0 +1,11 @@
+workflow:
+  as: baremetalds-metallb-e2e
+  steps:
+    pre:
+      - chain: baremetalds-ipi-pre
+    test:
+      - ref: baremetalds-metallb-e2e-test
```

Comment:
Why do you need this additional step? I wasn't expecting it, but to rely on the standard workflow. Everything you need should be driven by only `DEVSCRIPTS_CONFIG` (and in this case by adding the required config var `ENABLE_METALLB_MODE`), whereas the workflow should be default one

Reply:
e2e test is 'invoke e2e test' and it should not be a part of 'deployment of OCP cluster', as you did in https://github.com/openshift/release/blob/master/ci-operator/step-registry/baremetalds/e2e/baremetalds-e2e-workflow.yaml

If you still suppose we should everything is done in dev-scripts, then let's make another issue for that to invoke OpenShift-e2e and metallb-e2e from dev-scripts.

Reply:
(Minor point: it's not strictly required to define an additional step in the step-registry unless you plan to use the workflow in many different repos, you can save some burden by overriding it directly in the `tests` section)

Reply:
I'd expect it to run on a minimum of 2 repos -- `metallb` and `metallb-operator`.  We may want it as an optional job on `dev-scripts`, as well.

Reply:
@andfasano here's the PR that introduced the image build: https://github.com/openshift/release/pull/18375/files

Reply:
@andfasano Thank you for your reply!

1. One big reason is only dev-script image has packet server information (ssh-key and IP address for the server). packet server IP and auth information, such as ssh-key, is required to invoke metallb e2e tests because metallb e2e needs to access packet server to invoke e2e tests. Is there any good way to copy these information to src image?

That is why we ` source "${SHARED_DIR}/packet-conf.sh"` and `source "${SHARED_DIR}/ds-vars.conf"` here.

https://github.com/s1061123/release/blob/4dd292fea9506e442175b43661ad4df0257d7864/ci-operator/step-registry/baremetalds/metallb-e2e/test/baremetalds-metallb-e2e-test-commands.sh#L9-L17

2. As @dougbtv mentioned, metallb image is already built as 

https://github.com/openshift/release/blob/9122f1a1ff8ca7a429cdc67246344e6fb58939a4/ci-operator/config/openshift/metallb/openshift-metallb-main.yaml#L19-L26

Reply:
thanks @dougbtv

Reply:
> @andfasano Thank you for your reply!
> 
>     1. One big reason is only dev-script image has packet server information (ssh-key and IP address for the server). 

Secrets are not stored within the dev-scripts image (essentially the image just contains a copy of the source, see https://github.com/openshift/release/blob/bad39db84adafcad033a7060d7164eecb73815cf/ci-operator/config/openshift-metal3/dev-scripts/openshift-metal3-dev-scripts-master.yaml#L8), but they are extracted from the secrets management (ci-operator for the given cluster profile).
It's fine to source `packet-conf.sh` to retrieve them, as they're assembled there for the convenience of the job.

Not sure if your reply was referred to my previous comment (point 1 [here](https://github.com/openshift/release/pull/20007#discussion_r670574538)), in tthat case I was referring to the fact that the current `run_e2e.sh` script in dev-script in cloning the same `$METALLB_REPO` (`https://github.com/openshift/metallb.git`) that the Prow job _already_ checked out in the `src` image. This means that you could directly reuse it from the the step as you wish, without the need of cloning it again (ie, scp it to the remote Packet instance). But definitely that's an optimization you could address later

Reply:
@andfasano So which secret contains `packet-conf.sh` btw? I only know that is in dev-script image. Could you please let me know how to use it form `src`? then I'll copy metallb src to packet server (without clone) and try to invoke `run_e2e.sh` (`run_e2e.sh` can skip to clone if there is `metallb` already.

Reply:
@s1061123 regarding `packet-conf.sh` see the details [here](https://github.com/openshift/release/blob/ea8e908820aeadaca20c9ae9a2216ca65760dc52/ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh#L91-L108), but essentially it uses the provided ssh key found in the `$CLUSTER_PROFILE_DIR` (check this [doc](https://docs.ci.openshift.org/docs/architecture/step-registry/#available-environment-variables) for more details).

About using `src`, the CI really offers a number of different ways to adopt the prebuilt images into your tests (or build other ones) (see [Building Artifacts](https://docs.ci.openshift.org/docs/architecture/step-registry/#available-environment-variables)). In your case I guess that using `from: src` in the test step will be the simplest way,

Reply:
@andfasano changed.